src/libraries/Common/src/System/Net/Http/Http3/QPack/HeaderField.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+#if KESTREL
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http3.QPack
+#else
+namespace System.Net.Http.QPack
+#endif
+{
+    internal readonly struct HeaderField
+    {
+        public HeaderField(Span<byte> name, Span<byte> value)
+        {
+            Name = new byte[name.Length];
+            name.CopyTo(Name);
+
+            Value = new byte[value.Length];
+            value.CopyTo(Value);
```

Comment:
We just called Encoding.ASCII.GetBytes to get the byte[].  Why do we now need to allocate yet another set of arrays and copy into them?  Seems this should either just take strings, e.g.
```C#
public HeaderField(string name, string value)
{
   Name = Encoding.ASCII.GetBytes(name);
    Value = Encoding.ASCII.GetBytes(value);
}
```
or take the byte[] and just store them.

Reply:
Leaving as TODO because I don't know Kestrel uses this method.

Reply:
@jkotalik? I don't see any references outside of the static or dynamic tables.